### DISCUSSION TITLE
"Save as..." with "Download & Convert" Not Working

### DISCUSSION THREAD

**Discussion by Techno769** [aclap-dev/video-downloadhelper#163] [Help and Q&A]
### Your question

Hey there,
I've been using VDH for years and only just recently noticed that the add-on now automatically saves the files to the default selected download location without asking where to save, every time I use the "Download & Convert" function.
Essentially, the "Save as..." and "Save" buttons do exactly the same thing every time. Regardless of which option is ticked in settings.
I have tried reinstalling, verifying my license, reverting to default settings, and even using the Beta build with no luck.

Can someone look into this?

Also, I notice that every time I change the settings in the "Behavior" tab, the text becomes orange instead of black or green. Does this mean there is an error with the setting, or just to indicate it's been changed?

I did look, and couldn't find any other threads discussing this here.

Thanks

### Addon & CoApp version and details

version: 8.2.0.23a1
channel: beta
build date: 2024-02-14
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 8bf393bacf484e79
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.17",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Andrew",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e978"
}

**Comment by GrampaWildWilly** [comment]
Actually, somebody else did mention something about things turning orange.  Seems to me I asked that user for some screenshots.  But after a few searches in here, I'm not finding that discussion.  Maybe it will turn up when that user responds.  Meanwhile, you would help your cause if you posted a few screenshots showing what's happening to you.

**Reply by GrampaWildWilly** [reply]
He responded.  #155.  Not much help there.  Maybe you can advance the cause here.

**Comment by paulrouget** [comment]
orange means setting has changed.

You're correct, we do not respect the setting! I'll fix this in version 8.2.1.

**Reply by Techno769** [reply]
Thanks so much

**Reply by paulrouget** [reply]
This turns out to be a little more tricky than I thought.
I will release 8.2.1 first, and then look at that specific issue, and add a fix to the beta version.

**Reply by GrampaWildWilly** [reply]
Is orange meant to be an intermediate color?  I have a number of my settings that I have changed & they are all green.  I'm not really good with colors so I might never have noticed that anything was orange.  Does it change from black to orange when you change it, and then go to green when you click Save?

**Reply by GrampaWildWilly** [reply]
And is the bug here specific to Download and Convert?  Do other functions work fine?  What about a Download, and after that completes successfully, then do a Convert as a separate operation?

**Reply by paulrouget** [reply]
I think orange means "different than the default setting".

> And is the bug here specific to Download and Convert?

yes.

**Reply by GrampaWildWilly** [reply]
Well, I'm just not seeing any orange, only black & green.  Seriously.  I'm not trying to be a pest.  I do not see colors the way other people do.  I've had my eyes tested & I've been partially color blind all my life.  Are any of these things orange?  They all look either black or green to me.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6631335e-05fb-4abe-bdc1-342c5a5b4f8f)

**Reply by paulrouget** [reply]
You must be color blind :) There's no green in there. Only orange.

**Reply by GrampaWildWilly** [reply]
Pity me.  This is my world & I can't escape it.

**Reply by yeoldeman75** [reply]
Same here with the Save As  or Save and convert functions. No matter what I do with "Always Ask where to Save" it only saves to the default drive. This is a recent development within the last two weeks. Hopefully this will be fixed in the new version 8.2.1. Download speed has also dropped but that started a little longer ago and only with downloadhelper and no other apps.

**Reply by paulrouget** [reply]
@yeoldeman75 does this improve the situation for the download speed? https://github.com/aclap-dev/video-downloadhelper/discussions/199
